### DISCUSSION TITLE
Slicing xApp

### DISCUSSION THREAD

**Discussion by Higgigamer19** [srsran/srsRAN_Project#224] [General Help]
We were trying to deploy flexric with its slicing xApp, but ran into make errors in the code. We have determine a possibility that its the near real-time-ric that is causing the issue as its not sending the proper service model ID, the proper sm_id has a comment saying it hasn't been implemented yet. I wanted to ask if there were plans to finish it soon or do we need to implement it for ourselves? Also, is ran_function_ID and sm_id the same, because it seems they are used interchangeably?

The first error was the sent sm_id was not properly pass through the report_sm_xapp_api as shown, I believe the highlighted part is supposed to pass through the first element (0) and not the fourth (3). I captured the entire array as {147, 21298, 19744, 25956} where I noticed the first element was used in print statement for showing the ran_function_id in the terminal. ( flexric/examples/xApp/c/slice/xapp_slice_moni_ctrl.c  ;  line 262 )

Once I modified that, I ran into an error with sm_cb_slice where the _assert(rd->type == SLICE_STATS_V0)_ failed as _rd->type_ was supposed to have a value of 3 and it was actually 6. I believe this is due to the given sm_id being 147 not 145 changing the assigned rd.type value. ( flexric/examples/xApp/c/slice/xapp_slice_moni_ctrl.c  ;  line 41 in github, line 43 in photo )

I have screenshot and attached these lines, all errors are assertion errors within the called function or the line itself. 
![near_ric](https://github.com/srsran/srsRAN_Project/assets/77864957/da12e3d0-7b9a-4d8d-9374-7a4dbf3b8088)
![report_sm_xapp_api](https://github.com/srsran/srsRAN_Project/assets/77864957/fe8da241-7008-4eec-9fac-59017c6af670)
![sm_cb_slice](https://github.com/srsran/srsRAN_Project/assets/77864957/fad84e63-32c8-4d72-b2b0-178ca7dc5e24)

**Comment by pgawlowicz** [comment]
Hi, we use the Near-Real-Time RIC from the FlexRIC framework as an example RIC to control our gnb. Note that FlexRIC is a third-party project and we do not plan to extend it. 

Moreover, the E2 agent in our gnb does not support the SLICE RAN Function (here with id=145) as it is a custom function provided in FlexRIC framework. We plan to provide only standard-compliant E2 service modules, i.e., E2SM_KPM, E2SM_RC, E2SM_CCC. Note that the listed SMs provide functions to monitor and control slicing.
